ActorInspector is now backbone structure

Domily11 opened the pull request.

Domily11 committed:
modified actor model in joint.extentions

Added actortype parameter in joint.shapes.basic.Actor and placed a function to assign associated lines. Wouldn't show up without updating actorinspector

Domily11 committed:
Merge pull request #392 from amgrubb/mack-modifyactormodel

modified actor model in joint.extentions

Domily11 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Domily11 committed:
Implemented Backbone for ActorInspector

Actor Inspector is now as a backbone structure

Domily11 committed:
Fixed small bug of changeLine

Actor now has the associated line show up when you change the types

Domily11 committed:
Put placeholder in Actor input box

Domily11 committed:
Took off comments

leshghi commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -99,7 +99,7 @@ var Config = Backbone.View.extend({
 
```
Remove this file from commit

leshghi replied:
Remove this file from commit

leshghi commented on leaf-ui/rappid-extensions/joint.extensions.js:
```diff
@@ -223,9 +223,11 @@ joint.shapes.basic.Actor = joint.shapes.basic.Generic.extend({
     changeToOriginalColour: function() {
         this.attr({'.outer': {'fill': '#EBFFEA'}});
     }
+
```
Remove this line

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -50,12 +51,41 @@ var ActorInspector = Backbone.View.extend({
+        },
+
+        changeLine: function() {
+            console.log("changeLine");
```
Remove console.log

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -50,12 +51,41 @@ var ActorInspector = Backbone.View.extend({
+        },
+        
+        updateType: function(){
+            console.log("updateType");
```
Remove console.log

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -42,6 +42,7 @@ var ActorInspector = Backbone.View.extend({
          */
         nameAction: function(event) {
             // Prevent the ENTER key from being recorded when naming nodes.
+            console.log("nameAction");
```
Remove this console.log

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -16,23 +19,20 @@ var ActorInspector = Backbone.View.extend({
-            this.actor = model.getActorByID(cell.attributes.nodeID);
-
+        render: function() {
+            console.log("rendering");
```
Remove console.log

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -2,12 +2,15 @@
 
 var ActorInspector = Backbone.View.extend({
+        model: joint.shapes.basic.Actor,
+        //className: 'element-inspector', //change???
```
Remove this line for now

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -1176,7 +1178,7 @@ function clearInspector() {
 	linkInspector.clear();
 	analysisInspector.clear();
-	actorInspector.clear();
+	//actorInspector.clear();
```
Remove this line

leshghi replied:
This line still needs to be deleted

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -2,12 +2,15 @@
+        initialize: function() {
+            this.model.on('change:actorType', this.changeLine, this);
+        },
         template: [
```
You will want to update this template to be a script that takes in the model and autofills the inspector template with the model variables. I can help you with this on Monday. Take a look at ConfigInspector template in line 20 for an idea of how this works

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -16,23 +19,20 @@ var ActorInspector = Backbone.View.extend({
+        render: function() {
+            console.log("rendering");
             // If the clicked node is an actor, render the actor inspector
             this.$el.html(_.template(this.template)());
```
```suggestion
            this.$el.html(_.template($(this.template).html())(this.model.toJSON()));
```

leshghi commented on leaf-ui/rappid-extensions/joint.extensions.js:
```diff
@@ -223,9 +223,11 @@ joint.shapes.basic.Actor = joint.shapes.basic.Generic.extend({
         },
-        actor: new ActorTest()
+        actor: new ActorTest(),
+        actorType: 'A',
```
Great job adding actor type - we may also want to add the actorName as a parameter. Alternatively we could set up the code so that actor name and type go inside the actor test object. Let's discuss best options on Monday.

leshghi commented on leaf-ui/js/initializeElements.js:
```diff
@@ -84,7 +84,7 @@ var elementInspector = new ElementInspector();
 var configInspector = new ConfigInspector({collection:configCollection});
 var analysisInspector = new AnalysisInspector();
-var actorInspector =  new ActorInspector();
+
```
Remove the extra line here

leshghi commented on leaf-ui/js/initializeElements.js:
```diff
@@ -294,7 +294,7 @@ var commandManager = new joint.dia.CommandManager({ graph: graph });
 // A simple element editor.
 $('.inspector').append(elementInspector.el);
-$('.inspector').append(actorInspector.el);
+
```
Remove the extra line here

leshghi reviewed (changes requested):
Congrats you guys - this looks great as a start for the backbone refactor AND you added new functionality for the actor type! You should be very proud :). I am converting this to a draft just because we have a few bugs to iron out before checking it into the codebase. The two top ones are:
1. You will want to update this template to be a script that takes in the model and autofills the inspector template with the model variables. I can help you with this today but in the meantime take a look at ConfigInspector template in line 20 for an idea of how this works
2. We need to fix the issue where when you click the actor again/move the actor multiple actor inspector templates are showing up. I have some ideas for that (event listener created in the actorInspector view for clicks outside actor inspector to know to remove it from the side panel?) but no concrete solution yet. Feel free to do your own research/test out some things and we will meet and discuss later today :)

Awesome job again - its so cool to see this refactor taking shape! Thank you!

Domily11 committed:
Resolved some comments

Took out the console.logs and extra lines and unnecessary codes

Domily11 committed:
Took out extra line

Domily11 committed:
Template fixed

Domily11 committed:
Temporary fix for the name bug

The actor's name now appears on the text box every time you visit the actor inspector

Manarnz committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Fixed the inspector view problem

I connected the code with the changes Ali made and now it is working.

Domily11 committed:
Fix bug #408

Manarnz committed:
Added comments to the functions

To make reading the code easier, I added comments to the functions describing what each one does.

Manarnz committed:
Merge branch 'mack-actorinspector' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Revert "Added comments to the functions"

This reverts commit a4e8aff46b84659a57a562b87d4a902df466a989.

Manarnz committed:
Added Comments To Functions

leshghi commented on leaf-ui/js/initializeElements.js:
```diff
@@ -83,7 +83,7 @@ var linkInspector = new LinkInspector();
 var configCollection = new ConfigCollection([]);
 var configInspector = new ConfigInspector({collection:configCollection});
-var actorInspector =  new ActorInspector();
+//var actorInspector =  new ActorInspector();
```
You can delete this line

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -832,10 +832,11 @@ function createIntention(cell) {
 	var name = cell.attr('.name/text') + "_" + Actor.numOfCreatedInstances;
 	var actor = new Actor(name);
     cell.attr(".name/text", name);
+    cell.set('actorName', name);
```
You could potentially make a function inside the actor cell that updates the actorName and the .name/text param at the same time? So the function only has to be called once

Manarnz replied:
Is it necessary? Because we would only call it once.

Manarnz replied:
I decided not to create a function for that.

leshghi replied:
Okay, we may call it more times in the future (see createActor() function) but for now this is fine! We can always add a function if we end up using it a lot

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -2,37 +2,42 @@
-            this.$('.cell-attrs-text').val(this.actor.nodeName);
+            this.$el.html(_.template($(this.template).html())(this.model.toJSON()));
+            var actorType = this.model.get('actorType');
+            $('.actor-type').val(actorType);
```
You should be able to do this in the script as well - see lines 19-24 of AnalysisInspector.js

leshghi commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -99,7 +99,7 @@ var Config = Backbone.View.extend({
 
     /** Create and render dropdown inner view, set listeners */
```
Still need to remove this file from the commit - I can show you how if you want!

Manarnz replied:
Yes please show me!

leshghi replied:
I will show you rn

leshghi commented on leaf-ui/rappid-extensions/joint.extensions.js:
```diff
@@ -223,9 +223,12 @@ joint.shapes.basic.Actor = joint.shapes.basic.Generic.extend({
         this.attr({'.outer': {'fill': '#EBFFEA'}});
-    }
+    },
+    
```
Delete this line

leshghi reviewed (changes requested):
Some more changes to make and some stuff I mentioned last time that slipped through the cracks. Let me know if anything is confusing/you need help with it!

Manarnz committed:
Delete ConfigInspector.js from Commit

Manarnz committed:
Implemented small comments

Manarnz committed:
Merge branch 'mack-actorinspector' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

Manarnz committed:
Revert "Delete ConfigInspector.js from Commit"

This reverts commit ff3b24e828139df8e674fc6ce56599f95c171418.

Manarnz committed:
Changed the way the actorType placeholder updates

Before, the placeholder is updated in the render function in JS, now it is updated in the template itself in the script.

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -50,12 +52,46 @@ var ActorInspector = Backbone.View.extend({
             // Do not allow special characters in names, replace them with spaces.
             text = text.replace(/[^\w\n-]/g, ' ');
```
```suggestion
            var text = this.$('.cell-attrs-text').val().replace(/[^\w\n-]/g, ' ');
```

And delete line 51

leshghi commented on leaf-ui/rappid-extensions/ActorInspector.js:
```diff
@@ -50,12 +52,46 @@ var ActorInspector = Backbone.View.extend({
+         * Changes the line that distinguishes the type of actor 
+         */
+        changeLine: function() {
+            if (this.model.get('actorType') == 'G') {
```
Optional: Maybe a variable for this.model.get('actorType') instead of calling it 3 times?

leshghi commented on leaf-ui/rappid-extensions/joint.extensions.js:
```diff
@@ -223,9 +223,11 @@ joint.shapes.basic.Actor = joint.shapes.basic.Generic.extend({
             }
         },
-        actor: new ActorTest()
+        actor: new ActorTest(),
```
Add a TODO to replace with new ActorBBM once it is in codebase

leshghi reviewed (commented):
Once last round then you should be set

Manarnz committed:
implement Ali's *final* comments

leshghi committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into mack-actorinspector

leshghi reviewed (approved):
Removed ConfigInspector.js from pull request. You should be all set.
